packages/SidePanel/tests/SidePanel.spec.js

```diff
@@ -88,15 +87,96 @@ describe("SidePanel", () => {
+            <SidePanel.Overlay />
+          </SidePanel>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header>With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header kind="primary">With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+        </SidePanel.Group>
+      );
+      // data-pka-anchor
+      expect(getAllByTestId("heading")).toHaveLength(3);
+    });
+
+    it("throws the error when sidepanel has only one panel", () => {
+      const sidePanelGroup = () =>
+        renderReactTestingLibrary(
+          <SidePanel.Group>
+            <SidePanel onClose={noop} isOpen>
+              <SidePanel.Header>With Header</SidePanel.Header>
+              <SidePanel.Overlay />
+            </SidePanel>
+          </SidePanel.Group>
+        );
+      expect(sidePanelGroup).toThrow(Error);
+    });
+
+    it("should trigger onClick when clicking the x button", () => {
+      const noop = jest.fn();
+      const { getAllByTestId } = renderReactTestingLibrary(
+        <SidePanel.Group>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header>With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header>With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header kind="primary">With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+        </SidePanel.Group>
+      );
+      fireEvent.click(getAllByTestId(/button/i)[0]);
+      fireEvent.click(getAllByTestId(/button/i)[1]);
+      fireEvent.click(getAllByTestId(/button/i)[2]);
+
+      expect(noop).toBeCalledTimes(3);
+    });
+  });
+
   describe("SidePanel.Footer", () => {
     it("should included footer", () => {
       const fn = jest.fn();
       const { getByTestId } = render({
         onClose: fn,
-        children: <SidePanel.Footer data-pka-anchor="sidepanel.footer">Footer</SidePanel.Footer>,
+        children: <SidePanel.Footer>Footer</SidePanel.Footer>,
       });
 
       expect(getByTestId("sidepanel.footer")).toBeVisible();
     });
   });
+
+  describe("SidePanel.Trigger", () => {
+    it("renders with a default props", () => {
+      const defaultProps = {
+        children: <SidePanel.Trigger>Button</SidePanel.Trigger>,
+      };
+      const { getByRole } = render(defaultProps);
+      expect(getByRole(/button/)).toBeInTheDocument();
+    });
+
+    it("should tigger or render a button", () => {
+      const onClick = jest.fn();
+      const { getByRole } = render({
+        onClick,
+        children: <SidePanel.Trigger>Button</SidePanel.Trigger>,
+      });
+
+      expect(getByRole("button")).toBeInTheDocument();
```

Comment:
You have the same expect in line 169. Which means this test is testing the same thing as above.

Therefore, it makes this test kind of redundant. I think just keeping "renders with default props" is fine.

Reply:
done